File: 2_7/2_7.kt

fun findDestination(destinations: List<Pair<String, String>>): String {
    val map = mutableMapOf<String, String>()
    for ((from, to) in destinations) {
        if (from == to) {
            return "обратитесь к специалисту"
        }
        map[from] = to
    }

    var current = destinations[0].first
    while (map.containsKey(current)) {
        current = map[current]!!
    }
    return current
}

fun main() {
    val destinations = listOf(
        Pair("Новосибирск", "Дубай"),
        Pair("Новосибирск", "Искитим"),
        Pair("Искитим", "Линево")
    )
    println("Output: ${findDestination(destinations)}")
}

File: 2_7/2_7.cpp
#include <iostream>
#include <string>
#include <unordered_map>
#include <vector>
using namespace std;
string findDestination(vector<pair<string, string>> destinations) {
    unordered_map<string, string> map;
    for (const auto& p : destinations) {
        if (p.first == p.second) {
            return "обратитесь к специалисту";
        }
        map[p.first] = p.second;
    }
    
    string current = destinations[0].first;
    while (map.find(current) != map.end()) {
        current = map[current];
    }
    return current;
}

int main() {
    vector<pair<string, string>> destinations = { {"Новосибирск","Дубай"},{"Новосибирск","Искитим"},{"Искитим","Линево"}};
    cout << "Output: " << findDestination(destinations) << endl;
    
    return 0;
}